common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -17,31 +17,34 @@
 
 
 import java.util.List;
+import java.util.Objects;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 
-import org.junit.Ignore;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 
 public class LiveQueryTest extends BaseDbTest {
-    private static final long EXPECTED_DELAY_MS = 200;
-    private static final long SLOP_MS = EXPECTED_DELAY_MS / 10;
-    private static final long TOLERABLE_DELAY_MS = EXPECTED_DELAY_MS + SLOP_MS;
+    // query run time ranges from 1ms - 20ms depending on core,
+    // doubling it should guarantee query finishes running in tests everytime
+    private static final long QUERY_RUN_TIME = 20;
+    private static final long TOLERABLE_QUERY_RUN_TIME = QUERY_RUN_TIME * 2;
 
-    private static final String KEY = "number";
+    // when there's a rapid change within 250ms of previous query run,
+    // delay is 500ms + query run time. Add some fudge to guarantee we get callback from core
+    private static final long TOLERABLE_DATABASE_CHANGE_DELAY = 600;
```

Comment:
I think this should be written:
private static final long QUERY_DEBOUNCE_DELAY_MS = 500;

 "QUERY_DEBOUNCE_DELAY + QUERY_RUN_TIME + ((QUERY_DEBOUNCE_DELAY + QUERY_RUN_TIME) * 0.1)" or something.  Please show your work.  When something changes it needs to be easy for the next guy to figure out what to change.  If you add up all the numbers it is impossible to figure out what comes from the spec and what is your slop factor.
 
 Also, all time constants must include their units, in their names: _SEC, _MS, _MIN, etc